Source/Engine/Core/Math/Vector2.cs

```diff
@@ -1380,7 +1380,27 @@ public static Vector2 TransformNormal(Vector2 normal, Matrix transform)
             TransformNormal(ref normal, ref transform, out Vector2 result);
             return result;
         }
-
+        /// <summary>
+        /// Snaps Vector To Grid
+        /// </summary>
+        /// <param name="vector">Refrance vector</param>
+        /// <param name="GridSize">Grid cell size</param>
+        /// <returns></returns>
+        public static Vector2 SnapToGrid(Vector2 vector, Real GridSize)
+        {
+            vector -= (GridSize * 0.5f);
+            vector /= GridSize;
+            return Ceil(vector) * GridSize;
+        }
```

Comment:
This seems equivalent to 
```csharp
 public static Vector2 RoundTo(Vector2 vector, Real step)
        {
            return Round(vector / step) * step;
        }
```

Well, except for the very edge cases. `RoundTo(new Vector2(5, 0), 10)` will round up, while SnapToGrid will round down

Reply:
it is close but the SnapToGrid snaps to the center of grid cell and the offset make it consistent where the scap acures